File: 프로그래머스/lv0/120882. 등수 매기기/등수 매기기.py
def solution(score: list) -> list:
    dict, avg_list = {}, [sum(num) / 2 for num in score]

    for index, avg in enumerate(sorted(avg_list, reverse=True), start=1):
        if avg not in dict:
            dict[avg] = index

    return [dict[avg] for avg in avg_list]

File: 프로그래머스/lv0/120882. 등수 매기기/등수 매기기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<vector<int>> score) {
    vector<int> answer;
    vector<double> avg;
    for(auto v :score){
        avg.push_back(((double)v[1]+v[0])/2);
    }
    for(auto a : avg){
        int cnt = 1;
        for(auto b : avg){
            if(b>a){
                cnt+=1;
            }
        }
        answer.push_back(cnt);
    }
    return answer;
}